### ISSUE TITLE
Getting tracebacks from build_log.py

### ISSUE THREAD

**Issue by akesandgren** [hpcugent/easybuild-framework#2008]
I'm now and then getting this:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib/python2.7/logging/handlers.py", line 156, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 734, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
    msg = msg % self.args
TypeError: not enough arguments for format string
Logged from file build_log.py, line 149
```

**Comment by akesandgren** [comment]
I'm getting hit by it during build of ParaView 5.1.2 -mpi with intel 2017.01 which hits an internal compiler error on one of the files.

And i never get to see the actual build output, due to the traceback.

I have captured the ebmsg and args in build_log's error() function if it's of any use (quite large...)

**Comment by ocaisa** [comment]
I think this is an old problem with rolling over large log files, the log
file from ParaView is massive. From what I remember this doesn't affect the
installation, just the recording of the build output.

**Comment by boegel** [comment]
The related issue is #1094 , but the traceback is different here, this *may* be a different problem...

**Comment by boegel** [comment]
I've also seen this occurring with fairly small logs, e.g. with a failing HDF5 build.

My suspicion is that it only occurs when `%` is present in the error message, which is screwing up Python string templating done inside of `format` in the loggers...

**Comment by boegel** [comment]
fixed in #2024, let's follow up there
